service-data-access/src/main/java/gov/va/vro/abddataaccess/service/FhirClient.java

```diff
@@ -110,18 +123,37 @@ public String getUrl() {
    * @return a {@link Bundle}.
    * @throws AbdException when error occurs.
    */
-  public Bundle getBundle(AbdDomain domain, String patientIcn, int pageNo, int pageSize)
+  public Bundle getBundle(
+      AbdDomain domain, String patientIcn, String lighthouseToken, int pageNo, int pageSize)
       throws AbdException {
     SearchSpec searchSpec = domainToSearchSpec.get(domain).apply(patientIcn);
     String url = searchSpec.getUrl() + "&page=" + pageNo + "&count=" + pageSize;
-    String lighthouseToken = lighthouseApiService.getLighthouseToken(domain, patientIcn);
-    log.info("Get FHIR data from {}", url);
-    return client
-        .search()
-        .byUrl(url)
-        .returnBundle(Bundle.class)
-        .withAdditionalHeader(LIGHTHOUSE_AUTH_HEAD, lighthouseToken)
-        .execute();
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+    headers.set(LIGHTHOUSE_AUTH_HEAD, lighthouseToken);
+    HttpEntity request = new HttpEntity(headers);
+
+    String baseUrl = properties.lighthouseProperties().getFhirurl();
+    String fullUrl = baseUrl + "/" + url;
+    log.info("Get FHIR data from {}", fullUrl);
+    try {
+      ResponseEntity<String> response =
+          restTemplate.exchange(fullUrl, HttpMethod.GET, request, String.class);
+      if (response.getStatusCode() != HttpStatus.OK) {
+        log.error("Unexpected response from lighthouse {}", response.getStatusCode());
+        throw new AbdException("Unable to get bundle for the patient.");
+      }
+      String strBody = response.getBody();
+      Bundle bundle = jsonParser.parseResource(Bundle.class, strBody);
+      return bundle;
+    } catch (RestClientException ex) {
+      log.error("Unable to get bundle from {}", fullUrl);
```

Comment:
Could you also log the cause? log.error("Unable to get bundle from {}", fullUrl, ex); Same for next catch.